### PR TITLE
LB-1264: Releases by "Various Artists" should not be included on "Fresh releases" page

### PR DESCRIPTION
**Pull request by anshg1214**
This PR 
1. Adds filtering option for Various Artists in the frontend
2. Caches the API calls for fresh release page

### CODE REVIEW

**Comment by amCap1712** [comment]
This is not the right way to check for various artists. Various artists releases have the special artist_credit_id of 1.

**Comment by amCap1712** [comment]
I would suggest doing this filtering on frontend instead so that users have the option to show/hide va releases.

**Review by MonkeyDo — COMMENTED** [review]
Preliminary review/questions.

Maybe I'm misunderstanding some aspects here, let me know if I got it all wrong.

Haven't tried running the PR yet, I'm waiting until we merge the search PR to deploy it to test

File: frontend/js/src/explore/fresh-releases/FreshReleases.tsx
Comment (anshg1214) [reply frontend/js/src/explore/fresh-releases/FreshReleases.tsx:291]: I ran into a weird warning.
```
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
```

The issue was that while fetching, the dependencies of this useEffect releaseTypes and releaseTags are empty arrays, and they were being replaced by empty arrays. I added a debugger log this change, and it says [] is being replaced by [].

This is why I added a special case to handle this change

File: frontend/js/src/explore/fresh-releases/FreshReleases.tsx
Comment (anshg1214) [reply frontend/js/src/explore/fresh-releases/FreshReleases.tsx:297]: Correct.
I believe I found the issue I was facing. When I was creating a map for `allFilters` and sending to the ReleaseFilters component, But the issue was that, on every re-render, this map was being created again and again. And since it's a `map`, it is compared by reference and not by value like other non-primitive type data types, causing the useEffect to trigger.

File: frontend/js/src/explore/fresh-releases/FreshReleases.tsx
Comment (MonkeyDo) [inline frontend/js/src/explore/fresh-releases/FreshReleases.tsx:280]: I'm not sure how to test this locally properly, but I wonder if you need the stringification here.
Does this result int he same re-rendering loop issue still?
If so, I think something smells a bit fishy with the way we manipulate the state between parent and child components...
```
React.useEffect(() => {
    setFilteredList(releases);
  }, [releases]);
```

**Review by MonkeyDo — COMMENTED** [review]
Deployed to test.LB for testing, and working perfectly, nice work !

I do have a question below regarding the odd re-render loop you ran into previously, but feel free to merge if I'm just being thick :P

File: frontend/js/src/explore/fresh-releases/FreshReleases.tsx
Comment (anshg1214) [reply frontend/js/src/explore/fresh-releases/FreshReleases.tsx:280]: Yes, If i don't add a stringification, I get a warning on the browser console

`FreshReleases.tsx:283 Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.`

Comment (MonkeyDo) [reply frontend/js/src/explore/fresh-releases/FreshReleases.tsx]: I found the culprit !
I did say something was smelling fishy :D

In cases of circular dependency/render issues like this, my usual first step is to look at the dependency arrays and see if anything is missing in them.
In this case that turned out o be the issue, so an easy fix.
Disabling the linting rule `react-hooks/exhaustive-deps` is a dangerous game, it turns out :P

In our case, since we had `if (!_.isEqual(filteredReleases, filteredList)) {` and `filteredList` was never updated in the effect in ReleaseFilters, the condition was always true and was calling the parent's `setFilteredList`, which in turn re-rendered the child with the new `filteredList`, and so on, hence the infinite re-render loop.

**Review by MonkeyDo — APPROVED** [review]
With the circular dependency fixed, I think we are ready to go !
Nicely done :)

I will let you merge it yourself, in case there was something else you wanted ti look at or add.

File: frontend/js/src/explore/fresh-releases/FreshReleases.tsx
Comment (anshg1214) [reply frontend/js/src/explore/fresh-releases/FreshReleases.tsx]: Got it. Thank you so much for fixing this!

**Pull request merged**
